js/layout-javascript/agenda-code.js

```diff
@@ -1229,6 +1229,12 @@ DynamicList.prototype.renderLoopHTML = function () {
         _this.INCREMENTAL_RENDERING_BATCH_SIZE * (renderLoopIndex + 1)
       );
 
+      // Break render cycle after render to prevent rendering no results html
+      if (!nextBatch.length && renderLoopIndex > 0) {
```

Comment:
@MaksymShokin Can you explain the logic of why the problem occurred and why your change fixes the issue?

Reply:
@tonytlwu 
Basically we have the next flow.
Case 1 no results.
1. We pass first if statement. 
2. Render "no results" HTML
3. Return on the second if statement.

Case 2 3 results.
1. We pass first and second if statements 3 times and render 3 results.
2. On the fourth loop enter, we return on the first if statement without rendering "no results" HTML